azure-pipelines.release-fluentui.yml

```diff
@@ -1,74 +1,223 @@
+          StorageAccountRM: fluentsite
+          ContainerName: nightly-builds
+          # upload tarballs to folder named by today's date
+          BlobPrefix: $(date)
+          Gzip: true
+
+      - task: AzureUpload@2
+        displayName: '[Nightly] Upload tarballs as newest'
+        condition: and(succeeded(), eq(variables.publishOfficial, false))
+        inputs:
+          SourcePath: fluentui-nightly
+          ConnectedServiceNameARM: 4f791194-9458-4a18-9f9d-6a77ddb74b8c
+          StorageAccountRM: fluentsite
+          ContainerName: nightly-builds
+          # upload tarballs to folder named 'newest'. This folder is used by the nightly version of docsite to create codesandbox
+          BlobPrefix: newest
+          Gzip: true
+
+      - task: CmdLine@2
+        displayName: '[NPM] Publish to NPM'
+        condition: and(succeeded(), eq(variables.publishOfficial, true))
+        timeoutInMinutes: 2
+        inputs:
+          script: >-
+            yarn logout
+
+            export NPM_TOKEN=$(npmToken)
+
+            echo "@fluentui:registry=http://registry.npmjs.org/" > .npmrc
+
+            echo "registry=http://registry.npmjs.org/" >> .npmrc
+
+            echo //registry.npmjs.org/:_authToken=$(npmToken) >> .npmrc
+
+            touch packages/fluentui/.npmrc
+
+            echo "@fluentui:registry=http://registry.npmjs.org/" > packages/fluentui/.npmrc
+
+            echo "registry=http://registry.npmjs.org/" >> packages/fluentui/.npmrc
+
+            echo //registry.npmjs.org/:_authToken=$(npmToken) >> packages/fluentui/.npmrc
+
+            yarn release:fluentui:$(version) --yes
+
+            yarn release:fluentui:post-validation
+
+  - job: Job_2
+    displayName: Build and Publish Docsite
+    dependsOn: Job_1
+    # run this job when the previous job is succeeded or when publishDocsiteOnly is true
+    condition: or(succeeded(), eq(variables.publishDocsiteOnly, true))
+    pool:
+      name: Self Host Ubuntu
+
+    steps:
+      - checkout: self
+
+      - task: CmdLine@2
+        displayName: Authenticate Git credentials
+        inputs:
+          script: >-
+            git config user.name "Fluent UI Build"
+
+            git config user.email "fluentui-internal@service.microsoft.com"
+
+            git remote set-url origin https://$(githubUser):$(githubPAT)@github.com/microsoft/fluentui.git
+
+      - task: NodeTool@0
+        displayName: Install Node.js
+        inputs:
+          versionSpec: 12.x
+
+      - task: YarnInstaller@0
+        displayName: Use Yarn 1.19.x
+        inputs:
+          versionSpec: 1.19.x
+          checkLatest: true
+
+      - task: Bash@3
+        displayName: Yarn
+        inputs:
+          filePath: yarn-ci.sh
+
+      - task: CmdLine@2
+        displayName: 'Set base path and version'
+        condition: and(succeeded(), eq(variables.publishOfficial, true))
+        inputs:
+          script: >-
+            if [[ "$(publishOfficial)" == "true" ]]
```

Comment:
unnecessary condition ? this task will only run if `publishOfficial` is `true`

Reply:
Yeah you know, double security? :D If you think this would be confusing I could remove it

Reply:
it seems unnecessary to me, especially when the task is not even run, let's validate in a sane way :)